checkov/common/util/secrets.py

```diff
@@ -169,31 +168,34 @@ def omit_secret_value_from_checks(check: BaseCheck, check_result: dict[str, Chec
     return censored_code_lines
 
 
-def omit_secret_value_from_definitions(definitions: Dict[str, DictNode],
-                                       resource_attributes_to_omit: ResourceAttributesToOmit) -> Dict[str, DictNode]:
-    """
-        Mask secret values from definitions, as a way to mask these values in the created graph.
-        Should be used only in runners that have the resource_attributes_to_omit mapping
-        """
-    found_secrets = False
-    censored_definitions = definitions
-    for file, definition in definitions.items():
-        for i, resource in enumerate(definition.get('resource', [])):
-            for resource_type in [r_type for r_type in resource if r_type in resource_attributes_to_omit]:
-                for resource_name, resource_config in resource[resource_type].items():
-                    for attribute in [attribute for attribute in resource_config if
-                                      attribute in resource_attributes_to_omit[resource_type]]:
-                        if not found_secrets:
-                            found_secrets = True
-                            # The values in self.definitions shouldn't be changed so that checks' results
-                            # of checks that rely on the definitions values are not affected.
-                            # Hence, if secrets are found, we should censor them in a deep copy of self.definitions
-                            censored_definitions = copy.deepcopy(definitions)
-                        secret = resource_config[attribute][0]
-                        censored_value = omit_secret_value_from_line(secret, secret)
-                        censored_definitions[file]['resource'][i][resource_type][resource_name][attribute] = \
-                            [censored_value]
-    return censored_definitions
+def omit_secret_value_from_graph_checks(check: BaseGraphCheck, check_result: dict[str, CheckResult] | _CheckResult,
+                                        entity_code_lines: list[tuple[int, str]],
+                                        entity_config: dict[str, Any] | ParameterAttributes | ResourceAttributes,
+                                        resource_attributes_to_omit: ResourceAttributesToOmit | None = None) -> \
+        list[tuple[int, str]]:
+    secrets = set()  # a set, to efficiently avoid duplicates in case the same secret is found in the following conditions
+    censored_code_lines = []
+
+    if check.category == CheckCategories.SECRETS and check_result.get('result') == CheckResult.FAILED:  # type:ignore[comparison-overlap]
+        secrets.update([str(secret) for key, secret in entity_config.items() if
+                        key.startswith(f'{check.id}_secret')])
```

Comment:
```suggestion
        secrets = {
            str(secret) for key, secret in entity_config.items()
            if key.startswith(f'{check.id}_secret')
        }
```
this should do the job without needing the temporal list